core/base/src/TApplication.cxx

```diff
@@ -628,20 +630,423 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
+   TString Buff ("\\b");
+   Buff.Append(ClassName);
+   Buff.Append("::\\b");
+   TPRegexp ReturnFix(Buff.Data());
+   ReturnFix.Substitute(ReturnTypeName, "");
+   // We define the TString witch will be modified and returned from "InlineFunctionCheck".
+   TString ReturnType("");
+   // InlineCheck is not zero only if the function is defined as inline.
+   if (InlineCheck == 0) {
+      ReturnType.Append(ReturnTypeName);
+      // If the function is not inline we only change the spacing in "ReturnType".
+      ReturnType.ReplaceAll("*", " *");
+   } else {
+      // We check if the function is defined as virtual.
+      Int_t VirtualCheck = (((TMethod *)TClass::GetClass(ClassName)->GetListOfMethods()->FindObject(Method))->Property() & kIsVirtual);
+      if (VirtualCheck != 0) {
+         // If the function is virtual, we append "virtual" before the return type.
+         TString Buffer("virtual ");
+         Buffer.Append(ReturnTypeName);
+         ReturnType.Append(Buffer);
+      } else {
+         // If it's not virtual we only change the spacing in "ReturnType".
+         ReturnType.Append(ReturnTypeName);
+         ReturnType.ReplaceAll(" *", "*");
+      }
+   }
+   // In any case we need to change the return type as following.
+   ReturnType.ReplaceAll("istream", "std::istream");
+   ReturnType.ReplaceAll("ostream", "std::ostream");
+   ReturnType.ReplaceAll("&", " &");
+   return ReturnType;
+}
+}
+
+namespace {
+// A function that generates and loads in browser a URL for "Function" defined in "NameSpace".
+static void URLforFunction(TApplication &This, TString NameSpace, TString Function)
 {
-   gInterpreter->ProcessLine(line);
+   // The URL starts as a URL for namespace only (generated with "URLforNameSpace").
+   TString FinalURL(URLforNameSpace(NameSpace));
+   // Then "#a" is appended to the URL.
+   FinalURL.Append("#a");
+   // Then crypted with MD5 text is appended. The structure of the texst is as follow: 
+   // "ReturnType NameSpace::FunctionFunction(function arguments)".
+   TString Check(DoxygenSyntax(NameSpace, Function));
+   TString ReturnType("");
+   if(Check.EqualTo("()")){
+      ReturnType.Append("constexpr ");
+   }
+   ReturnType.Append(((TFunction *)(((TMethod *)(TClass::GetClass(NameSpace.Data()))->GetListOfMethods())->FindObject(Function.Data())))->GetReturnTypeName());
+   ReturnType.Append(" ");
+   TString MD5Text(ReturnType.Data());
+   MD5Text.Append(NameSpace.Data());
+   MD5Text.Append("::");
+   MD5Text.Append(Function.Data());
+   MD5Text.Append(Function.Data());
+   // We use "DoxygenSyntax" (it works the same way for namespaces) to modify the arguments of 
+   // Method with respect of Doxygen.
+   MD5Text.Append(DoxygenSyntax(NameSpace, Function));
+   // We crypt the text with MD5 - method in TString.
+   FinalURL.Append(MD5Text.MD5());
+   // With "OpenInBrowser" we load the generated URL in a browser.
+   This.OpenInBrowser(FinalURL);
```

Comment:
I would move this call `OpenInBrowser` into the caller, returning only the `finalURL`. That way you don't need to take `TApplication` as parameter anymore.